src/test/TestMarket.cpp

```diff
@@ -72,11 +72,25 @@ operator==(OfferState const& x, OfferState const& y)
     return true;
 }
 
-ClaimOfferAtom
-TestMarketOffer::exchanged(int64_t sold, int64_t bought) const
+ClaimAtom
+TestMarketOffer::exchanged(uint32_t ledgerVersion, int64_t sold,
+                           int64_t bought) const
 {
-    return ClaimOfferAtom{key.sellerID, key.offerID,  state.selling,
-                          sold,         state.buying, bought};
+    ClaimAtom atom;
+    if (ledgerVersion < 18)
```

Comment:
What do you think about putting this logic into a helper that returns the right claim atom based on the ledger version? It could be used in `crossOffer` and `crossOfferV10` as well.

Reply:
That's cleaner. Done.